concrete_types.hh

```diff
@@ -19,6 +19,7 @@
 #include "utils/big_decimal.hh"
 
 struct empty_type_impl final : public abstract_type {
+    using native_type = empty_type_representation;
```

Comment:
empty_type is also used in materialized views, in some cases, for unselected columns where we need to remember their liveness but not their value. See commit 30f721afabb4c0f8bf7c3c32e0ab02bcb36b77e4 - "virtual columns". This empty_type does get written to sstables. We should check that it doesn't get broken by this patch. We have tests for the virtual columns case - but it is very likely that none of them ever tried to flush to sstables and then read without cache and see that it still works....

Reply:
I think the series doesn't change any of that. The path impacted is conversion to the native type (e.g. int32_t for int32_type), I think views use the serialized form throughout. I'll check it of course.